### PR TITLE
Completed unit tests jest

### PR DESCRIPTION
**Pull request by eric-k-chu**
Capture

### CODE REVIEW

File: unit-tests-jest/numbers.test.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    expect(result).toEqual('$35.00');
+  });
+
+  it('returns float round-down converted to dollars', () => {
+    const number = 46.23178;
+    const result = toDollars(number);
+    expect(result).toEqual('$46.23');
+  });
+
+  it('returns float one decimal converted to dollars', () => {
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.ts:36]: I like this test. I didn't think of doing it myself, but it's a great edge case to test!

File: unit-tests-jest/numbers.test.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    expect(result).toEqual([]);
+  });
+
+  it('returns a new array', () => {
+    const numbers: number[] = [];
+    const copy = numbers.map((n) => n);
+    const divisor = 5;
+    const result = divideBy(numbers, divisor);
+    expect(result).toEqual([]);
+    expect(numbers).toEqual(copy);
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.ts:64]: It's cool you're testing this, but I don't think this test actually tests that the input and output arrays are different. Did you verify that the test fails if divideBy simply modifies and returns the input?

If `numbers` had a value in it, your code would do a better job of testing. But instead, you might try `.toBe` which will test for reference equality. For example, something like `expect(numbers).not.toBe(result)`.

File: unit-tests-jest/numbers.test.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  });
+});
+
+describe('multiplyBy', () => {
+  it('object is modified, where values are multiplied by multiplier', () => {
+    const obj = { a: 5, b: 1, c: 3 };
+    multiplyBy(obj, 3);
+    expect(obj).toEqual({ a: 15, b: 3, c: 9 });
+  });
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.ts:73]: Again, good thing to check. This looks like it works. My test again used `toBe` to check reference equality.

**Review by RobertGardner — APPROVED** [review]
Nice. A fun comment for you to experiment with.
